packages/devextreme/js/__internal/core/state_manager/setup_state_manager.ts

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable spellcheck/spell-checker */
+import type { DIContext } from '../di';
+import { Logger } from './logger';
+import { StateManagerFactory } from './state_manager';
+import type * as StateManagementTypes from './types';
+
+export type DecoratorFunction<T = unknown> = (instance: T) => T;
+
+export interface StateManagerInitializerOptions {
+  logLevel?: StateManagementTypes.LogLevel;
+  componentName?: string;
+  diContext?: DIContext;
+  controllerSign?: string;
+}
+
+const CONTROLLER_SIGN = 'Controller';
+
+function isController(
+  instance: unknown,
+  controllerSign: string,
+): instance is StateManagementTypes.StateSource {
+  if (instance) {
+    return typeof instance === 'object'
+         && 'constructor' in instance
+         && 'name' in instance.constructor
+         && instance.constructor.name.includes(controllerSign);
+  }
+
+  return false;
```

Comment:
[Minor comment]

Let's make it shorter, if instance is casted to false (except null) -> condition return false anyway
So, we can make it shorter without splitting part of condition in if operator:
```suggestion
    return instance !== null // or !!instance to make it 100% the same condition
         && typeof instance === 'object'
         && 'constructor' in instance
         && 'name' in instance.constructor
         && instance.constructor.name.includes(controllerSign);
```

Reply:
changed